opslevel/resource_opslevel_team_contact.go

```diff
@@ -1,146 +1,211 @@
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	contactCreateInput := opslevel.ContactInput{
+		Address:     data.Value.ValueString(),
+		DisplayName: data.Name.ValueStringPointer(),
+		Type:        opslevel.ContactType(data.Type.ValueString()),
+	}
+
+	teamIdentifier := data.Team.ValueString()
+
+	contact, err := teamContactResource.client.AddContact(teamIdentifier, contactCreateInput)
+	if err != nil || contact == nil {
+		resp.Diagnostics.AddError("opslevel client error", fmt.Sprintf("unable to add contact on team (%s), got error: %s", teamIdentifier, err))
+		return
+	}
+
+	createdTeamContactModel := NewTeamContactResourceModel(*contact)
+	createdTeamContactModel.Team = RequiredStringValue(teamIdentifier)
+	createdTeamContactModel.LastUpdated = timeLastUpdated()
+	tflog.Trace(ctx, "created a team contact resource")
+	resp.Diagnostics.Append(resp.State.Set(ctx, &createdTeamContactModel)...)
+}
+
+func (teamContactResource *TeamContactResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var data TeamContactResourceModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	teamIdentifier := data.Team.ValueString()
+	contactID := data.Id.ValueString()
+
+	var team *opslevel.Team
+	var err error
+	if opslevel.IsID(teamIdentifier) {
+		team, err = teamContactResource.client.GetTeam(opslevel.ID(teamIdentifier))
+	} else {
+		team, err = teamContactResource.client.GetTeamWithAlias(teamIdentifier)
+	}
+	if err != nil || team == nil {
+		resp.Diagnostics.AddError("opslevel client error", fmt.Sprintf("unable to read team (%s), got error: %s", teamIdentifier, err))
+		return
+	}
+	err = team.Hydrate(teamContactResource.client)
+	if err != nil {
+		resp.Diagnostics.AddError("opslevel client error", fmt.Sprintf("unable to hydrate team (%s), got error: %s", teamIdentifier, err))
+	}
+
+	var teamContact *opslevel.Contact
+	for _, readContact := range team.Contacts {
+		if string(readContact.Id) == contactID {
+			teamContact = &readContact
+			break
+		}
+	}
+	if teamContact == nil {
+		resp.Diagnostics.AddError("opslevel client error", fmt.Sprintf("team contact (with ID '%s') not found on team (%s)", contactID, teamIdentifier))
+		return
+	}
+
+	readTeamContactResourceModel := NewTeamContactResourceModel(*teamContact)
+	readTeamContactResourceModel.Team = RequiredStringValue(teamIdentifier)
+	resp.Diagnostics.Append(resp.State.Set(ctx, &readTeamContactResourceModel)...)
+}
+
+func (teamContactResource *TeamContactResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var data TeamContactResourceModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	contactCreateInput := opslevel.ContactInput{
+		Address:     data.Value.ValueString(),
+		DisplayName: data.Name.ValueStringPointer(),
```

Comment:
```suggestion
		DisplayName: opslevel.RefOf(data.Name.ValueString()),
```
I found this subtle bug today. For Update only, this change in code will support unsetting this optional field

Reply:
`data.Name.ValueStringPointer(),` will return `nil` and opslevel-go will ignore this field. But we actually want to unset it here so we need to provide a string pointer to a blank string

Reply:
@davidbloss done